linker/resolve.go

```diff
@@ -477,7 +380,7 @@ func (r *result) resolveFieldTypes(handler *reporter.Handler, s *Symbols, fqn pr
 				// need to validate that it's not an illegal reference. To be valid, the field
 				// must be repeated and the entry type must be nested in the same enclosing
 				// message as the field.
-				isValid = dsc.FullName() == fqn && fld.GetLabel() == descriptorpb.FieldDescriptorProto_LABEL_REPEATED
+				isValid = dsc.FullName() == f.FullName() && fld.GetLabel() == descriptorpb.FieldDescriptorProto_LABEL_REPEATED
```

Comment:
This logic is not right. I've just updated `fqn` to extract the fully-qualified name. But this will never be true. It should be trying to verify that the field and the message belong to the same parent message. I will address in a subsequent PR as well as add a test that would have caught this.